lib/llm/tests/test_jail.rs

```diff
@@ -337,7 +380,8 @@ mod tests {
             .build();
 
         let jailed_stream = jail.apply(input_stream);
-        let results: Vec<_> = jailed_stream.collect().await;
+        let fixed_stream = JailedStream::fix_finish_reason(jailed_stream);
```

Comment:
For non-tool calling tests this `fix_finish_reason` isn't actually relevant currently, is it? Or are there other edge cases where we need to fix the finish reason?